highfidelity/hifi
# El Boppo the Clown punching bag game

El Boppo the Clown is a game in which you have a punching bag in the shape of a clown. The game comes with a timer, scoreboard and sound effects.

In this PR a feature has been added to the teleport.js script:
  -  you can now ignore an entity in the teleport.js script, which allows you to teleport while having an entity attached that covers your hands in such a way it would normally block the teleport laser. 
     - To add to the ignore list:
        ```Messages.sendLocalMessage('Hifi-Teleport-Ignore-Add', 'YourEntityIDHere');```
     - To remove from the ignore list:
        ```Messages.sendLocalMessage('Hifi-Teleport-Ignore-Remove', 'YourEntityIDHere');```
## QA test plan
 For this test you need an HMD and hand-controllers:
 - Use the build of this PR ( because changes to teleport.js have been made )
 - Run the following script to create the El Boppo game: ```https://rawgit.com/thoys/hifi/bdabc21d57a65a218961fa42c050da147c79b3e2/unpublishedScripts/marketplace/boppo/createElBoppo.js```
 - While entering the zone and HMD is enabled, the boxing gloves will auto-attach to your hands.
 - Teleporting should still work while the boxing gloves are being worn.
 - To start, give the punching bag an initial punch, you should hear a bell and the game music should start now, as well as the timer will start running.
 - Every punch should add a point to the score.
 - In the last 3 seconds you should hear a ticking sound. (each second)
 - When the game ends you should hear the end bell, and the music should stop.
